src/battle/battle_controller_player.c

```diff
@@ -84,10 +85,563 @@ void BattleControllerPlayer_PokemonAppear(BattleSystem *bsys, BATTLECONTEXT *ctx
+                    continue;
+                }
+
+                if (ov12_02261264(BattleSystem_GetOpponentDataByBattlerId(bsys, battlerIdCheck)) != 0x1) {
+                    continue;
+                }
+
+                if (ctx->unk_0[battlerIdCheck] != 14) {
+                    break;
+                }
+            }
+
+            if (battlerIdCheck == battlersMax) {
+                ov12_02262B80(bsys, ctx, battlerId, ctx->selectedMonIndex[battlerId]);
+                ctx->unk_0[battlerId] = 1;
+            } else {
+                break;
+            }
+        //fallthrough
+        case 1:
+            if (ov12_0225682C(ctx, battlerId)) {
+                ctx->unk_21A8[battlerId][3] = ctx->unk_2300[battlerId][0];
+
+                if (battleType & BATTLE_TYPE_PAL_PARK) {
+                    switch (ov12_0225682C(ctx, battlerId)) {
+                    case 1:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 17;
+                        break;
+                    case 4:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 16;
+                        break;
+                    }
+                } else if (battleType & BATTLE_TYPE_SAFARI) {
+                    switch (ov12_0225682C(ctx, battlerId)) {
+                    case 1:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 17;
+                        break;
+                    case 2:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 18;
+                        break;
+                    case 3:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 19;
+                        break;
+                    case 4:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_4[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 16;
+                        break;
+                    case 5:
+                        ctx->unk_0[battlerId] = 17;
+                        ctx->unk_0[battlerId] = 13;
+                        ctx->unk_21A8[battlerId][0] = 20;
+                        break;
+                    }
+                } else {
+                    if (ov12_0225682C(ctx, battlerId) != 0xff) {
+                        ctx->unk_314C[battlerId] |= 0x1;
+                    }
+
+                    switch (ov12_0225682C(ctx, battlerId)) {
+                    case 1: //Fight
+                        if (StruggleCheck(bsys, ctx, battlerId, 0, 0xffffffff) == 15) {
```

Comment:
nope, that field is the stuggle check flags, so it's basically saying check for all of them here (they just included 0xFFFFFFFF instead of a concat of all of them for some reason)